### PR TITLE
ABAP Unit for ZCL_ABAPGIT_BRANCH_OVERVIEW #1647

### PR DESCRIPTION
**Pull request by JohannesKonings**
https://github.com/larshp/abapGit/issues/1647

Preparation for ABAP Unit. Next step would be creating an interface for https://github.com/larshp/abapGit/blob/master/src/zcl_abapgit_repo_online.clas.abap to mock a online repo.

### CODE REVIEW

**Comment by christianguenter2** [comment]
Can you check the commits? Seem the same issue as in https://github.com/larshp/abapGit/pull/1655

File: src/zcl_abapgit_factory.clas.abap [hunk 1]
```diff
@@ -154,4 +170,32 @@ CLASS zcl_abapgit_factory IMPLEMENTATION.
 
   ENDMETHOD.
 
+  METHOD get_branch_overview.
+
+    DATA: ls_branch_overview LIKE LINE OF gt_branch_overview,
+          lv_branch          TYPE ty_branch_overview-branch.
+    FIELD-SYMBOLS: <ls_branch_overview> TYPE zcl_abapgit_factory=>ty_branch_overview.
+
+    lv_branch = io_repo->get_sha1_remote( ).
```
Comment (christianguenter2) [inline src/zcl_abapgit_factory.clas.abap]: Maybe we should use the repo key instead?

**Comment by JohannesKonings** [comment]
I've "reorganized" the commits and changed from branch to repo key

**Comment by larshp** [comment]
one note, the title of this PR is "ABAP Unit for ZCL_ABAPGIT_BRANCH_OVERVIEW" however it does not add any ABAP Unit? Just introduces an interface for the class

File: src/zcl_abapgit_factory.clas.abap [hunk 2]
```diff
@@ -154,4 +170,32 @@ CLASS zcl_abapgit_factory IMPLEMENTATION.
...
+  METHOD get_branch_overview.
+
+    DATA: ls_branch_overview LIKE LINE OF gt_branch_overview,
+          lv_repo_key        TYPE ty_branch_overview-repo_key.
+    FIELD-SYMBOLS: <ls_branch_overview> TYPE zcl_abapgit_factory=>ty_branch_overview.
+
+    lv_repo_key = io_repo->get_key( ).
+
+    READ TABLE gt_branch_overview ASSIGNING <ls_branch_overview>
+                               WITH TABLE KEY repo_key = lv_repo_key.
```
Comment (larshp) [inline src/zcl_abapgit_factory.clas.abap]: this looks dangerous, the repo_key ids are reused, if a repo is deleted and another added, then it gets the same key, and io_repo is stored in the singleton.

Suggest not making this a singleton, instead use an internal table with injections for unit testing

**Comment by JohannesKonings** [comment]
You're right it is not consistent. It's the title of the issue but didn't fit to the commits.
I've changed the title of the PR.

File: src/zcl_abapgit_factory.clas.abap (hunk 2 above)
Comment (JohannesKonings) [reply src/zcl_abapgit_factory.clas.abap]: I corrected it in https://github.com/larshp/abapGit/pull/1665
The implementation of your suggestion I will do later.

Comment (christianguenter2) [reply src/zcl_abapgit_factory.clas.abap]: Ah, good to know. Then it wasn't the best idea.

Comment (larshp) [reply src/zcl_abapgit_factory.clas.abap]: it is also not the best idea the way it finds the keys 😄 I just dont like singletons, they sometimes introduce evil bugs

**Pull request merged**